docs/development/sync-protocol.md

```diff
@@ -0,0 +1,431 @@
+| clean_sync | NO | bool | true |
+
+
+Example Preflight request JSON Payload:
+
+```json
+{
+  "compiler_rule_count" : 14,
+  "client_mode" : "MONITOR",
+  "santa_version" : "2022.6",
+  "serial_num" : "XXXZ30URLVDQ",
+  "binary_rule_count" : 43676,
+  "hostname" : "markowsky.example.com",
+  "primary_user" : "markowsky",
+  "certificate_rule_count" : 2364,
+  "teamid_rule_count" : 0,
+  "os_build" : "21F5048e",
+  "fcm_token" : "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXy88D1MWVn3c79waWewilH4vhK8TV1FMQn8dncj4XekvKuFhhvFCNNdI0gc02-IIx_lRGgUkhPmwNBE_T32h6LYQbR5U04aZY_Q0",
+  "transitive_rule_count" : 0,
+  "os_version" : "12.4",
+  "model_identifier" : "MacBookPro15,1",
+  "clean_sync": true,
+}
+```
+
+#### `preflight` Response 
+
+If all of the data is well formed, the server responds with a HTTP 200 and provides a JSON response with the 
+When a 200 is returned by the server it has a JSON object in the response.
+
+This returns a JSON object of with the following keys:
+
+| Key | Required | Type | Meaning | Example Value |
+|---|---|---|---|---|
+| enable_bundles | NO | boolean | Enabled bundle scanning  | true |
+| bundles_enabled | NO | boolean |  **DEPRECATED** Enable bundle scanning (use `enable_bundles`)  | true |
+| enable_transitive_rules | NO | boolean | Whether or not to enable transitive allowlisting | true |
+| batch_size | YES | integer | Number of events to upload at a time | 128 | 
+| fcm_full_sync_interval | NO | integer | Number of seconds between full syncs | 14400 |
+| fcm_global_rule_sync_deadline | NO | unsigned integer | How often to download new rules from the sync server in seconds | 600 |
+| full_sync_interval | YES | integer | Number of seconds between full syncs | 600 | 
+| client_mode | YES | string | Operating mode to set for the client | either "MONITOR" or "LOCKDOWN" |
+| allowed_path_regex | YES | list of strings | List of regular expressions to allow a binary to execute from a path | ["/Users/markowsk/foo/.*"] | 
+| blocked_path_regex | YES | list of strings | List of regular expressions to block a binary from executing by path | ["/tmp/"] |
+| block_usb_mount | NO | boolean | Block usb mass storage devices | true |
+| remount_usb__mode | NO | string |  |
+| clean_sync | YES | boolean | Whether or not the rules should be dropped and synced entirely from the server | true |
+
+#### Example Preflight Response Payload
+
+```json
+{
+ "batch_size": 100, 
+ "client_mode": "MONITOR", 
+ "allowed_path_regex": null, 
+ "blocked_path_regex": null, 
+ "clean_sync": false, 
+ "fcm_full_sync_interval": 14400, 
+ "fcm_global_rule_sync_deadline": 600,
+ "bundles_enabled": true, 
+ "enable_transitive_rules": false
+}
+```
+
+### Event Upload
+
+After the `preflight` stage has completed the client then initiates the
+`eventupload` stage if it has any events to upload. If there aren't any events
+this stage is skipped.
+
+It consists of the following transaction, that may be repeated until all events are uploaded..
```

Comment:
call out that the number of events sent in a single request is controlled by the batch_size value in the preflight response